trin-history/src/jsonrpc.rs

```diff
@@ -50,6 +51,27 @@ impl HistoryRequestHandler {
                         };
                     let _ = request.resp.send(response);
                 }
+                HistoryEndpoint::Store => {
+                    let response = match StoreParams::<HistoryContentKey>::try_from(request.params)
+                    {
+                        Ok(params) => {
+                            let content_key = params.content_key.clone();
+                            let content = params.content.clone();
+                            match self
+                                .network
+                                .overlay
+                                .storage
+                                .write()
+                                .store(&content_key, &content)
+                            {
+                                Ok(_) => Ok(Value::Bool(true)),
+                                Err(_) => Ok(Value::Bool(false)),
+                            }
+                        }
+                        Err(_) => Ok(Value::Bool(false)),
```

Comment:
I'm not sure that we should return `false` on an error. It would be useful to the client to know why a particular call failed. For example, the target node's storage may be full.

Reply:
If we always propagate the error to the client, then we would never return `false`, so you could argue that a `Bool` result type does not make sense, but I think `Bool` is better than `String`. I can't think of a case where we would return `false` and the client would not be interested in the reason for failure.

Reply:
Given that this is a dev or debug endpoint, and that the API client would likely be a dev, I think it would be useful to have some information if some store request does not succeed. We could log the error to surface that information, but that is only useful if the client (dev) is running `trin` locally.

Reply:
> I can't think of a case where we would return `false` and the client would not be interested in the reason for failure.

This is a good point. I think it makes sense to return an error to the client if something fails and we are already doing it, for example, if we add more arguments than expected or if we are unable to decode the arguments.

Not sure what is better for `portal_*Store` specifications thought, we can just say something like: "You SHOULD return `false` on an error and you MAY return the error code/message too". I think it is easier to standardize this way and we leave the error message as "nice to have" implementation detail.

So in our case, I'm more in favor to return `boolean` plus an error message if available as `Value::Array`.  This way, we match the specs and provide an additional error message.

How does this sound?

Reply:
Tbh, since this is a "dev" endpoint, I don't feel strongly that we need to match the spec. I do agree that returning the error message is useful. 

> So in our case, I'm more in favor to return boolean plus an error message if available as Value::Array. This way, we match the specs and provide an additional error message.

I'd disagree here. This suggestion doesn't match the spec. The spec dictates that we return a single bool, true or false. But again, I don't feel strongly that we need to match the spec.

Imo, decision 1 is do we need to match the spec?
- Yes - Ok let's update the spec & collab with fluffy to flesh out the ideal return type
- No - Ok, what does our return type look like? 
  - I don't feel strongly here either, but I'd probably lean towards just returning a string. "true" for success, "{error_message}" for failure.

Reply:
I wasn't aware that the endpoint was in the spec. My main concern was that a dev would try to call this endpoint to store some data and it fails with no reason available to the dev.